<a href="https://colab.research.google.com/github/matiasdemori/Trabajo-Integrador-Ciencia-de-Datos/blob/main/TP_Integrados_Ciencia_de_Datos.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# TRABAJO PRÁCTICO GRUPAL INTEGRADOR
*Unidad:* 1 y 2

*Materia:* Ciencia de Datos  


---


*Fecha Entrega 1:* 27/04/2026

*Fecha Entrega 2:* 11/05/2026

### Integrantes del grupo:
* Franco Lavirgen
* Matias De Mori
* Juan Cruz Palacio Hernandez
* Angelo Grammatico

### Docentes:
* Germán Giró
* Malena Constancio
* Giselle Bustamante


# 1ra Entrega
En esta primera entrega buscamos definir el problema que vamos a analizar, comprender el contexto del proyecto y realizar una exploración inicial del dataset seleccionado siguiendo las primeras etapas de la metodología CRISP-DM. Además, planteamos nuestras preguntas analíticas principales, justificamos la elección del dataset y evaluamos de manera preliminar la estructura y calidad de los datos con los que vamos a trabajar.

##Comprensión del negocio

### Contexto
El precio de los combustibles representa una variable de gran impacto dentro de la economía argentina, ya que influye directamente en los costos de transporte, logística, producción y distribución de bienes y servicios. Además, las variaciones de precios afectan tanto a consumidores particulares como a empresas y actividades comerciales en distintas regiones del país.
Debido a esto, resulta relevante analizar cómo se comportan los precios de los combustibles según distintos factores como la ubicación geográfica, la empresa comercializadora y el tipo de producto ofrecido.

###Problema
Si bien existe información pública y actualizada sobre precios de combustibles en estaciones de servicio de Argentina, el volumen de datos y la frecuencia de actualización dificultan la identificación de patrones claros sin un análisis previo.
El problema principal consiste en comprender qué factores explican las diferencias de precios observadas en el mercado y determinar si ciertas brechas de precios se mantienen estables en el tiempo o presentan fluctuaciones significativas.

###Objetivos
<u>_Objetivo principal_</u>

- Analizar en qué medida el factor geográfico, la bandera comercial y el tipo de combustible influyen en la variabilidad de los precios promedio semanales en Argentina durante los últimos 12 meses.

<u>_Objetivos secundarios_</u>


- Comparar precios promedio semanales entre provincias.
- Analizar la dispersión de precios entre distintas empresas comercializadoras.
- Evaluar diferencias de precios entre tipos de combustible.
- Detectar regiones o empresas con precios sistemáticamente más altos o más bajos.
- Observar la estabilidad o fluctuación de las brechas de precios a lo largo del período analizado.



### Preguntas a Analizar
<u>_Pregunta analítica principal_</u>

¿En qué medida el factor geográfico, la bandera comercial y el tipo de producto explican la variabilidad de los precios semanales en Argentina, y qué tan estables se mantienen las brechas de mercado entre estos segmentos a lo largo del último año?

<u>_Preguntas analíticas secundarias_</u>

*Segmentación geográfica*

- ¿Qué provincias presentan precios promedio semanales por encima de la media nacional y qué tan amplia es la brecha entre la región más cara y la más barata?

*Competencia entre empresas*

- ¿Cómo varía la dispersión de los precios promedio semanales entre las diferentes empresas del sector, y qué proporción de marcas mantienen precios por encima de la mediana nacional de forma constante?

*Análisis por tipo de combustible*

- ¿Cuál es la magnitud de la brecha de precio entre los distintos tipos de combustible (Premium vs. Súper) y cómo ha fluctuado esta diferencia porcentual en el promedio semanal de los últimos 12 meses?
Alcance del análisis



##Comprensión de los datos

###Dataset seleccionado

- Precios en Surtidor – Resolución 314/2016
Fuente oficial: Portal de Datos Abiertos de la República Argentina.

- Contiene información sobre precios de combustibles relevados en estaciones de servicio del país.

El conjunto de datos posee una cantidad muy elevada de registros, ya que contiene información histórica desde años anteriores y actualizaciones frecuentes. Por este motivo, trabajar la base completa excede el alcance del trabajo práctico.

<u>_**Recorte definido para el análisis**_</u>

Para hacer el proyecto manejable y enfocado, se decidió trabajar con:

- Los últimos 12 meses disponibles del dataset.
- Precio promedio semanal como unidad de análisis.

Esto permite reducir el volumen de datos, disminuir ruido generado por actualizaciones horarias y enfocarse en tendencias recientes.


### Importacion de librerias necesarias
- <u>_pandas_</u>: se utilizará para leer, organizar, limpiar y analizar datos en forma de tablas, similares a una planilla de Excel.
- <u>_matplotlib.pyplot_</u>: se utilizará para generar gráficos y visualizar la información del dataset de una manera más clara.
- <u>_urllib.request_</u>: se utilizará para acceder o descargar archivos desde una URL de internet, por ejemplo un dataset en formato CSV.

In [28]:
import pandas as pd
import matplotlib.pyplot as plt
import urllib.request

- <u>_url_</u>: almacena la dirección web donde se encuentra el archivo CSV que vamos a utilizar.
- <u>_pd.read_csv(url)_</u>: lee el archivo CSV directamente desde internet.
- <u>_df_</u>: es el DataFrame donde quedan guardados los datos del archivo para poder analizarlos luego.

In [29]:
url = "http://datos.energia.gob.ar/dataset/1c181390-5045-475e-94dc-410429be4b17/resource/f8dda0d5-2a9f-4d34-b79b-4e63de3995df/download/precios-historicos.csv"

df = pd.read_csv(url)

###Describir la estructura del dataset y análisis preliminar de calidad:

Observamos las 10 primeras filas del dataset para darnos cuenta que hay caracteres del tipo "." o " " los cuales necesitan limpieza al igual que la columna *fecha_vigencia* que necesita estar apta para cambiar a *dtype:fecha*

In [30]:
df.head(10)

,idempresa,empresa,direccion,localidad,provincia,idproducto,producto,idtipohorario,tipohorario,precio,fecha_vigencia,idempresabandera,empresabandera,latitud,longitud,geojson,anio,mes
0,1376,10 DE SETIEMBRE S.A.,Av. Mosconi 299,LOMAS DEL MIRADOR,BUENOS AIRES,19,Gas Oil Grado 2,2,Diurno,985.0,01/04/2024 10:11,28,PUMA,-34.658476,-58.529443,"{""type"":""Point"",""coordinates"":[-58.529443,-34....",2024,4
1,1376,10 DE SETIEMBRE S.A.,Av. Mosconi 299,LOMAS DEL MIRADOR,BUENOS AIRES,19,Gas Oil Grado 2,3,Nocturno,985.0,01/04/2024 10:11,28,PUMA,-34.658476,-58.529443,"{""type"":""Point"",""coordinates"":[-58.529443,-34....",2024,4
2,1376,10 DE SETIEMBRE S.A.,Av. Mosconi 299,LOMAS DEL MIRADOR,BUENOS AIRES,19,Gas Oil Grado 2,2,Diurno,2190.0,01/05/2026 12:43,28,PUMA,-34.658476,-58.529443,"{""type"":""Point"",""coordinates"":[-58.529443,-34....",2026,5
3,1376,10 DE SETIEMBRE S.A.,Av. Mosconi 299,LOMAS DEL MIRADOR,BUENOS AIRES,19,Gas Oil Grado 2,3,Nocturno,2190.0,01/05/2026 12:43,28,PUMA,-34.658476,-58.529443,"{""type"":""Point"",""coordinates"":[-58.529443,-34....",2026,5
4,1376,10 DE SETIEMBRE S.A.,Av. Mosconi 299,LOMAS DEL MIRADOR,BUENOS AIRES,19,Gas Oil Grado 2,2,Diurno,1080.0,01/07/2024 09:43,28,PUMA,-34.658476,-58.529443,"{""type"":""Point"",""coordinates"":[-58.529443,-34....",2024,7
5,1376,10 DE SETIEMBRE S.A.,Av. Mosconi 299,LOMAS DEL MIRADOR,BUENOS AIRES,19,Gas Oil Grado 2,3,Nocturno,1080.0,01/07/2024 09:43,28,PUMA,-34.658476,-58.529443,"{""type"":""Point"",""coordinates"":[-58.529443,-34....",2024,7
6,1376,10 DE SETIEMBRE S.A.,Av. Mosconi 299,LOMAS DEL MIRADOR,BUENOS AIRES,19,Gas Oil Grado 2,2,Diurno,1108.0,01/08/2024 13:40,28,PUMA,-34.658476,-58.529443,"{""type"":""Point"",""coordinates"":[-58.529443,-34....",2024,8
7,1376,10 DE SETIEMBRE S.A.,Av. Mosconi 299,LOMAS DEL MIRADOR,BUENOS AIRES,19,Gas Oil Grado 2,3,Nocturno,1108.0,01/08/2024 13:40,28,PUMA,-34.658476,-58.529443,"{""type"":""Point"",""coordinates"":[-58.529443,-34....",2024,8
8,1376,10 DE SETIEMBRE S.A.,Av. Mosconi 299,LOMAS DEL MIRADOR,BUENOS AIRES,19,Gas Oil Grado 2,2,Diurno,172.5,01/10/2022 10:23,28,PUMA,-34.658476,-58.529443,"{""type"":""Point"",""coordinates"":[-58.529443,-34....",2022,10
9,1376,10 DE SETIEMBRE S.A.,Av. Mosconi 299,LOMAS DEL MIRADOR,BUENOS AIRES,19,Gas Oil Grado 2,3,Nocturno,172.5,01/10/2022 10:23,28,PUMA,-34.658476,-58.529443,"{""type"":""Point"",""coordinates"":[-58.529443,-34....",2022,10


El codigo a continuacion nos permite conocer los nombres de las columnas, la cantidad de valores no nulos en cada una y el tipo de dato de cada variable.

Esto lo utilizaremos para  describir la estructura inicial del dataset e identificar posibles problemas, como columnas con valores faltantes, variables numéricas guardadas como texto o campos que podrían necesitar una transformación antes del análisis.

In [31]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3342323 entries, 0 to 3342322
Data columns (total 18 columns):
 #   Column            Dtype  
---  ------            -----  
 0   idempresa         int64  
 1   empresa           object 
 2   direccion         object 
 3   localidad         object 
 4   provincia         object 
 5   idproducto        int64  
 6   producto          object 
 7   idtipohorario     int64  
 8   tipohorario       object 
 9   precio            float64
 10  fecha_vigencia    object 
 11  idempresabandera  int64  
 12  empresabandera    object 
 13  latitud           float64
 14  longitud          float64
 15  geojson           object 
 16  anio              int64  
 17  mes               int64  
dtypes: float64(3), int64(6), object(9)
memory usage: 459.0+ MB


Observamos que el dataset posee 18 columnas y 3342322 filas, en base a nuestras preguntas analiticas y la orientacion que le vamos a dar al proyecto solo vamos a usar las 5 columnas:

- Empresabandera
- Precio
- Producto
- Provincia
- Fecha_vigencia

En las cuales Fecha_vigencia es del
tipo object por lo tanto necesitaremos cambiar el dtype mas adelante:

Contamos la cantidad de valores nulos o vacíos que tiene cada columna del dataset.

Esto nos sirve para realizar un primer análisis de calidad de los datos, ya que nos permite identificar qué variables tienen información incompleta. A partir de este resultado podremos decidir si es necesario limpiar datos, completar valores faltantes, eliminar registros o descartar alguna variable que tenga demasiada información ausente.

In [32]:
df.isnull().sum()

,0
idempresa,0
empresa,0
direccion,0
localidad,0
provincia,0
idproducto,0
producto,0
idtipohorario,0
tipohorario,0
precio,0


Generamos un resumen estadístico de las columnas numéricas del dataset.

Esto nos sirve para analizar de manera preliminar el comportamiento de las variables cuantitativas. A partir de este resumen podemos observar valores como la media, el mínimo, el máximo y los distintos cuartiles.

También nos ayuda a detectar posibles inconsistencias o valores atípicos, por ejemplo precios demasiado altos, valores iguales a cero, registros negativos o diferencias muy grandes entre el mínimo y el máximo. Esta información es útil para definir qué variables podrían necesitar limpieza, validación o transformación antes de avanzar con el análisis.


In [33]:
df.describe()


,idempresa,idproducto,idtipohorario,precio,idempresabandera,latitud,longitud,anio,mes
count,3.342323e+06,3.342323e+06,3.342323e+06,3.342323e+06,3.342323e+06,3.340861e+06,3.340861e+06,3.342323e+06,3.342323e+06
mean,5.073572e+03,1.099478e+01,2.499988e+00,4.580390e+02,9.469797e+00,-3.352593e+01,-6.147591e+01,2.021449e+03,6.440122e+00
std,3.197128e+03,8.636993e+00,5.000001e-01,2.722729e+03,1.092387e+01,4.095867e+00,3.621483e+00,1.665761e+01,3.628330e+00
min,1.300000e+01,2.000000e+00,2.000000e+00,0.000000e+00,-1.000000e+00,-5.482950e+01,-7.233893e+01,1.000000e+00,1.000000e+00
25%,2.066000e+03,3.000000e+00,2.000000e+00,4.699000e+01,2.000000e+00,-3.479213e+01,-6.423593e+01,2.019000e+03,3.000000e+00
50%,4.852000e+03,6.000000e+00,2.000000e+00,1.374000e+02,4.000000e+00,-3.432265e+01,-6.056582e+01,2.022000e+03,6.000000e+00
75%,7.932000e+03,1.900000e+01,3.000000e+00,8.610000e+02,2.600000e+01,-3.146485e+01,-5.853706e+01,2.024000e+03,1.000000e+01
max,1.145300e+04,2.100000e+01,3.000000e+00,2.430000e+06,3.000000e+01,-2.205582e+01,-5.364826e+01,9.920000e+03,1.200000e+01


*describe* nos deja ver aspectos como\ *mean, min, max* de nuestras variables numericas, lo cual usaremos como herramienta en proximas etapas.

Calculamos tanto la cantidad como el porcentaje de valores faltantes en cada columna del dataset.

Esto nos sirve para evaluar la calidad inicial de los datos y detectar qué variables presentan información incompleta. A partir de este análisis podremos decidir, en etapas posteriores, si conviene eliminar registros, imputar valores, conservar las variables o descartarlas en caso de que tengan demasiados datos faltantes.

In [34]:
# Calculamos la cantidad y el porcentaje de valores faltantes por columna
valores_faltantes = pd.DataFrame({
    "cantidad_faltantes": df.isnull().sum(),
    "porcentaje_faltantes": df.isnull().mean() * 100
})

valores_faltantes.sort_values("porcentaje_faltantes", ascending=False)

,cantidad_faltantes,porcentaje_faltantes
geojson,1462,0.043742
longitud,1462,0.043742
latitud,1462,0.043742
idempresa,0,0.000000
empresa,0,0.000000
direccion,0,0.000000
producto,0,0.000000
localidad,0,0.000000
provincia,0,0.000000
idproducto,0,0.000000


Con este código verificamos si existen filas duplicadas dentro del dataset. Esto lo hacemos para realizar un control preliminar de calidad, ya que los registros repetidos podrían distorsionar los resultados del análisis. Si encontráramos duplicados, en una etapa posterior deberíamos evaluar si corresponde eliminarlos o conservarlos según el significado de cada registro.

In [35]:
df.duplicated().sum()

np.int64(0)

Cantidad de valores únicos por variable

Obvservamos cuántos valores distintos tiene cada columna del dataset.Para diferenciar variables categóricas, numéricas, geográficas o identificatorias. También nos ayuda a detectar posibles inconsistencias, por ejemplo categorías escritas de distintas maneras, columnas con demasiados valores únicos o variables que podrían necesitar agrupación o limpieza.


In [36]:
df.nunique().sort_values(ascending=False)

,0
fecha_vigencia,354946
precio,19794
idempresa,5698
direccion,5611
geojson,5035
latitud,5023
longitud,5022
empresa,4161
localidad,1146
anio,62


Vemos los valores únicos de las variables categóricas más importantes para nuestro análisis, como provincia, empresa y tipo de combustible.

Esto es para conocer las categorías disponibles y detectar posibles inconsistencias en la escritura de los datos. Además, nos permite validar si estas variables son adecuadas para responder nuestras preguntas analíticas relacionadas con diferencias geográficas, comerciales y por tipo de producto.

In [37]:
print(df["provincia"].unique())
print(df["empresa"].unique())
print(df["producto"].unique())

['BUENOS AIRES' 'RIO NEGRO' 'CHUBUT' 'SAN LUIS' 'SANTA FE' 'ENTRE RIOS'
 'CHACO' 'SANTIAGO DEL ESTERO' 'MENDOZA' 'CATAMARCA' 'CORDOBA' 'SALTA'
 'MISIONES' 'TUCUMAN' 'CORRIENTES' 'LA PAMPA' 'LA RIOJA' 'CAPITAL FEDERAL'
 'SANTA CRUZ' 'NEUQUEN' 'JUJUY' 'FORMOSA' 'SAN JUAN' 'TIERRA DEL FUEGO']
['10 DE SETIEMBRE S.A.' '1263 S.R.L.' '1913 SRL' ... 'ZOKFA S.A.S.'
 'ZONA CERO S.A' 'ZUSICO S.A.']
['Gas Oil Grado 2' 'Gas Oil Grado 3' 'GNC'
 'Nafta (premium) de más de 95 Ron' 'Nafta (súper) entre 92 y 95 Ron']


In [38]:
df["empresabandera"].value_counts()

,count
empresabandera,
YPF,1225658
SHELL C.A.P.S.A.,772886
AXION,522830
PUMA,318790
BLANCA,297213
DAPSA S.A.,73099
GULF,60526
REFINOR,33742
VOY,18013


Al observar los valores que toma esta columna nos damos cuenta de que blanca puede tomar dos valores:

Blanca
Sin empresa bandera
Podemos ver que hay empresas que pueden estar duplicadas ya que tuvieron un cambio de nombre, por ejemplo Puma y Petrobras.

In [39]:
df["producto"].value_counts()

,count
producto,
Nafta (súper) entre 92 y 95 Ron,802061
Gas Oil Grado 2,797300
Gas Oil Grado 3,791662
Nafta (premium) de más de 95 Ron,779144
GNC,172156


Al observar los valores de esta columna no vemos ninguna incongruencia.

In [40]:
df["provincia"].value_counts()

,count
provincia,
BUENOS AIRES,1284697
CORDOBA,407312
SANTA FE,329733
CAPITAL FEDERAL,164794
ENTRE RIOS,154852
MENDOZA,123448
RIO NEGRO,81416
MISIONES,78601
TUCUMAN,77536


**Podemos** observar viendo los valores de esta columna que se encuentran las 23 provincias y capital federal, sin duplicados.

###**Justificacion del dataset**

Se eligió este dataset porque presenta características adecuadas para responder las preguntas analíticas planteadas en el proyecto y permite trabajar distintas dimensiones relacionadas con la variabilidad de los precios de combustibles en Argentina.

Motivos principales:
* Relevancia práctica

  El precio de los combustibles representa una variable de gran impacto económico, ya que influye directamente en transporte, logística, inflación y costos operativos en distintas regiones del país.

  Analizar cómo varían los precios según la provincia, la empresa comercializadora y el tipo de combustible permite comprender mejor el comportamiento del mercado y detectar diferencias entre segmentos.

Relación con las preguntas analíticas

Las variables seleccionadas:

* *Empresa bandera*
* *Precio*
* *Producto*
* *Provincia*
* *Fecha vigencia*

Estas variables permiten abordar directamente la pregunta analítica principal del proyecto, enfocada en estudiar cómo influyen el factor geográfico, la bandera comercial y el tipo de combustible sobre la variabilidad de los precios semanales.

Además, estas variables permiten desarrollar las preguntas secundarias vinculadas a:

* Diferencias regionales entre provincias.
* Dispersión de precios entre empresas.
* Brechas de precios entre distintos tipos de combustible.

#2da Entrega

##Preparación de los datos

Realizamos un recorte temporal del dataset, tomando como referencia abril de 2026 y seleccionando los registros correspondientes a un año hacia atrás.

Este filtro fue la segmentacion planteada para el recorte de datos, haciendo que el análisis sea más manejable y esté alineado con el alcance definido para el trabajo. Además, nos permite comparar los precios de combustibles dentro de un intervalo temporal concreto, evitando mezclar datos históricos demasiado antiguos con datos actuales.

In [45]:
# Nos aseguramos de que la columna de fecha esté en formato datetime
df["fecha_vigencia"] = pd.to_datetime(df["fecha_vigencia"], errors="coerce")

# Definimos el rango de fechas que queremos analizar
fecha_inicio = "2025-04-01"
fecha_fin = "2026-04-30"

# Filtramos el dataset para quedarnos solo con los registros entre abril 2025 y abril 2026
df_recorte = df[
    (df["fecha_vigencia"] >= fecha_inicio) &
    (df["fecha_vigencia"] <= fecha_fin)
].copy()

# Verificamos el tamaño del nuevo dataset filtrado
df_recorte.shape

(351473, 18)

Con este código verificamos cuál es la fecha mínima y máxima del dataset recortado.

Esto nos sirve para confirmar que el filtro temporal se aplicó correctamente y que estamos trabajando únicamente con registros comprendidos entre abril de 2025 y abril de 2026.

In [46]:
df_recorte["fecha_vigencia"].min(), df_recorte["fecha_vigencia"].max()

(Timestamp('2025-04-01 00:00:00'), Timestamp('2026-04-30 00:00:00'))

Con este código agrupamos los registros del dataset recortado por mes.

Esto nos sirve para analizar si la cantidad de datos está distribuida de manera relativamente pareja entre abril de 2025 y abril de 2026, o si existen meses con mayor o menor cantidad de registros.

In [47]:
# Creamos una columna con el año y mes de cada registro
df_recorte["anio_mes"] = df_recorte["fecha_vigencia"].dt.to_period("M")

# Contamos la cantidad de registros por mes
registros_por_mes = df_recorte["anio_mes"].value_counts().sort_index()

# Mostramos el resultado
registros_por_mes

,count
anio_mes,
2025-04,29817
2025-05,48086
2025-06,41497
2025-07,22339
2025-08,33023
2025-09,27277
2025-10,20871
2025-11,33726
2025-12,16596


En esta etapa limpiamos el dataset quedándonos únicamente con las columnas que vamos a utilizar en nuestro análisis.

Este recorte nos sirve para simplificar el dataset, eliminar columnas que no serán utilizadas y trabajar de manera más ordenada sobre las variables directamente relacionadas con nuestras preguntas analíticas.

In [55]:
# Seleccionamos únicamente las columnas necesarias para nuestro análisis
columnas_seleccionadas = [
    "empresabandera",
    "precio",
    "producto",
    "provincia",
    "fecha_vigencia"
]

# Creamos un nuevo DataFrame solo con esas columnas
df_limpio = df_recorte[columnas_seleccionadas].copy()

# Mostramos las primeras filas del nuevo dataset
df_limpio.head()

,empresabandera,precio,producto,provincia,fecha_vigencia
10,PUMA,1733.0,Gas Oil Grado 2,BUENOS AIRES,2026-02-02
11,PUMA,1733.0,Gas Oil Grado 2,BUENOS AIRES,2026-02-02
16,PUMA,2182.0,Gas Oil Grado 2,BUENOS AIRES,2026-04-02
17,PUMA,2182.0,Gas Oil Grado 2,BUENOS AIRES,2026-04-02
22,PUMA,1237.0,Gas Oil Grado 2,BUENOS AIRES,2025-06-03


In [59]:
df_limpio["fecha_vigencia"].min(), df_limpio["fecha_vigencia"].max()

(Timestamp('2025-04-01 00:00:00'), Timestamp('2026-04-30 00:00:00'))

In [60]:
df_limpio.columns

Index(['empresabandera', 'precio', 'producto', 'provincia', 'fecha_vigencia'], dtype='object')

### Tratamiento de datos faltantes

Se realizó un análisis de valores faltantes sobre las variables seleccionadas para esta segunda etapa del trabajo práctico.

In [61]:
# Creamos una tabla con cantidad y porcentaje de valores faltantes
faltantes_df_limpio = pd.DataFrame({
    "cantidad_faltantes": df_limpio.isnull().sum(),
    "porcentaje_faltantes": df_limpio.isnull().mean() * 100
})

# Mostramos la tabla ordenada de mayor a menor porcentaje de faltantes
faltantes_df_limpio.sort_values("porcentaje_faltantes", ascending=False)

,cantidad_faltantes,porcentaje_faltantes
empresabandera,0,0.0
precio,0,0.0
producto,0,0.0
provincia,0,0.0
fecha_vigencia,0,0.0


Con este código buscamos registros donde el precio sea menor o igual a 100.

Esto nos sirve para identificar posibles inconsistencias, ya que en el contexto del análisis un precio de combustible igual a 100 o negativo no tendría sentido económico. Si aparecieran estos casos, deberían ser revisados y posiblemente excluidos o tratados en la etapa de limpieza.

In [80]:
cantidad_precios_negativos = (df_limpio["precio"] <= 100).sum()

cantidad_precios_negativos

np.int64(8)

Generamos un resumen de las columnas categóricas del dataset.

Esto nos sirve para conocer cuántos valores no nulos tiene cada variable, cuántas categorías distintas existen y cuál es la categoría más frecuente. Esta información es útil para comprender mejor variables como provincia, empresa o tipo de combustible, que son centrales para nuestras preguntas analíticas.

In [66]:
df_limpio.describe(include="object")

,empresabandera,producto,provincia
count,351473,351473,351473
unique,10,5,24
top,SHELL C.A.P.S.A.,Nafta (súper) entre 92 y 95 Ron,BUENOS AIRES
freq,117522,88618,148139


Luego del análisis realizado, se observó que las variables utilizadas no presentan valores faltantes.

Por este motivo, no fue necesario aplicar técnicas de imputación, eliminación de registros ni reemplazo de datos.

Esta decisión permite conservar la integridad del dataset original y trabajar únicamente sobre las transformaciones necesarias para el análisis exploratorio.

###Corrección de formatos y tipos de datos.

Con este código transformamos la columna de fecha al formato datetime de Pandas.

Esto nos sirve porque nuestro análisis se enfocará en la evolución temporal de los precios durante los últimos 12 meses. Para poder filtrar períodos, agrupar por semana y analizar tendencias, necesitamos que la fecha esté correctamente reconocida como una variable temporal.

In [67]:
df_limpio["fecha_vigencia"] = pd.to_datetime(
    df_limpio["fecha_vigencia"],
    errors="coerce"
).dt.normalize()

In [68]:
df_limpio.dtypes

,0
empresabandera,object
precio,float64
producto,object
provincia,object
fecha_vigencia,datetime64[ns]


In [69]:
# Normalizamos columnas categóricas eliminando espacios al inicio y al final
columnas_texto = ["empresabandera", "producto", "provincia"]

for columna in columnas_texto:
    df_limpio[columna] = df_limpio[columna].astype(str).str.strip()

##Analisis Descriptivo

En esta sección analizamos la variable `precio`, que es la principal variable numérica del dataset. Calculamos medidas de tendencia central y dispersión para conocer el comportamiento general de los precios de combustibles durante el período seleccionado.

In [81]:
df_limpio["precio"].describe()

,precio
count,351473.000000
mean,1620.748164
std,324.352634
min,0.000000
25%,1407.000000
50%,1587.000000
75%,1825.000000
max,3000.000000


Calculamos de forma separada la media, mediana, desvío estándar, valor mínimo y valor máximo del precio.

Esto nos sirve para interpretar con mayor claridad el nivel promedio de precios, su variabilidad y la diferencia entre los valores extremos del dataset.

In [82]:
# Calculamos medidas principales de la variable precio
media_precio = df_limpio["precio"].mean()
mediana_precio = df_limpio["precio"].median()
desvio_precio = df_limpio["precio"].std()
min_precio = df_limpio["precio"].min()
max_precio = df_limpio["precio"].max()

print("Media del precio:", media_precio)
print("Mediana del precio:", mediana_precio)
print("Desvío estándar del precio:", desvio_precio)
print("Precio mínimo:", min_precio)
print("Precio máximo:", max_precio)

Media del precio: 1620.7481644393738
Mediana del precio: 1587.0
Desvío estándar del precio: 324.35263446345687
Precio mínimo: 0.0
Precio máximo: 3000.0


Analizamos la cantidad y el porcentaje de registros correspondientes a cada tipo de combustible.

Esto nos permite conocer qué productos tienen mayor presencia en el dataset y evaluar si la distribución de registros entre categorías es equilibrada o si algún tipo de combustible está más representado que otro.

In [83]:
# Calculamos frecuencia absoluta y relativa por producto
frecuencia_producto = pd.DataFrame({
    "cantidad": df_limpio["producto"].value_counts(),
    "porcentaje": df_limpio["producto"].value_counts(normalize=True) * 100
})

frecuencia_producto

,cantidad,porcentaje
producto,,
Nafta (súper) entre 92 y 95 Ron,88618,25.213317
Gas Oil Grado 2,87031,24.761788
Gas Oil Grado 3,85482,24.321072
Nafta (premium) de más de 95 Ron,82675,23.522433
GNC,7667,2.181391


Analizamos la cantidad y el porcentaje de registros por provincia.

Esto nos sirve para revisar la representación geográfica del dataset y detectar si algunas provincias concentran una mayor cantidad de observaciones que otras.

In [84]:
frecuencia_provincia = pd.DataFrame({
    "cantidad": df_limpio["provincia"].value_counts(),
    "porcentaje": df_limpio["provincia"].value_counts(normalize=True) * 100
})

frecuencia_provincia

,cantidad,porcentaje
provincia,,
BUENOS AIRES,148139,42.148046
CORDOBA,42128,11.986127
SANTA FE,30361,8.638217
CAPITAL FEDERAL,21098,6.002737
ENTRE RIOS,16931,4.817155
MENDOZA,9229,2.625806
RIO NEGRO,8287,2.357791
CORRIENTES,7892,2.245407
TUCUMAN,7642,2.174278


Analizamos la cantidad y el porcentaje de registros por empresa comercializadora.

Esto nos permite identificar qué empresas tienen mayor presencia en el dataset y evaluar si existe una concentración de registros en determinadas banderas.

In [85]:
frecuencia_empresa = pd.DataFrame({
    "cantidad": df_limpio["empresabandera"].value_counts(),
    "porcentaje": df_limpio["empresabandera"].value_counts(normalize=True) * 100
})

frecuencia_empresa

,cantidad,porcentaje
empresabandera,,
SHELL C.A.P.S.A.,117522,33.436992
YPF,70652,20.101686
AXION,68692,19.544033
PUMA,43886,12.486308
BLANCA,21976,6.252543
DAPSA S.A.,11238,3.197401
GULF,10859,3.089569
REFINOR,3285,0.934638
VOY,3207,0.912446


Buscamos posibles valores atípicos en la variable `precio` utilizando el método del rango intercuartílico.

Esto nos sirve para detectar precios que se encuentran muy alejados del comportamiento general de los datos. Estos casos pueden representar errores de carga, situaciones excepcionales o diferencias reales entre productos, empresas o provincias.

In [86]:
# Calculamos los cuartiles y el rango intercuartílico
Q1 = df_limpio["precio"].quantile(0.25)
Q3 = df_limpio["precio"].quantile(0.75)
IQR = Q3 - Q1

# Definimos los límites para detectar valores atípicos
limite_inferior = Q1 - 1.5 * IQR
limite_superior = Q3 + 1.5 * IQR

print("Límite inferior:", limite_inferior)
print("Límite superior:", limite_superior)

# Filtramos los registros considerados valores atípicos
outliers_precio = df_limpio[
    (df_limpio["precio"] < limite_inferior) |
    (df_limpio["precio"] > limite_superior)
]

outliers_precio

Límite inferior: 780.0
Límite superior: 2452.0


,empresabandera,precio,producto,provincia,fecha_vigencia
82,PUMA,2469.0,Gas Oil Grado 3,BUENOS AIRES,2026-04-02
83,PUMA,2469.0,Gas Oil Grado 3,BUENOS AIRES,2026-04-02
130,PUMA,599.9,GNC,BUENOS AIRES,2026-04-02
131,PUMA,599.9,GNC,BUENOS AIRES,2026-04-02
136,PUMA,489.9,GNC,BUENOS AIRES,2025-05-05
...,...,...,...,...,...
3339763,SHELL C.A.P.S.A.,639.0,GNC,CORDOBA,2025-05-01
3339794,SHELL C.A.P.S.A.,598.0,GNC,CORDOBA,2025-07-22
3339795,SHELL C.A.P.S.A.,598.0,GNC,CORDOBA,2025-07-22
3341731,PUMA,689.0,GNC,BUENOS AIRES,2025-08-06


Calculamos cuántos registros fueron identificados como valores atípicos y qué porcentaje representan sobre el total del dataset.

Esto nos ayuda a evaluar el impacto potencial de estos valores en el análisis. Si el porcentaje es bajo, podrían analizarse como casos particulares; si es alto, habría que revisar si responden a diferencias reales del mercado o a problemas de calidad de datos.

In [87]:
cantidad_outliers = outliers_precio.shape[0]
porcentaje_outliers = cantidad_outliers / df_limpio.shape[0] * 100

print("Cantidad de valores atípicos:", cantidad_outliers)
print("Porcentaje de valores atípicos:", porcentaje_outliers)

Cantidad de valores atípicos: 7947
Porcentaje de valores atípicos: 2.261055614513775
